### PR TITLE
Fix find multiple tags

### PR DESCRIPTION
**Pull request by seanflyyy**
This PR will allow for the following
* find t/python t/beginner will return the students with both `python` and `beginner` tags at minimum

(@cadencjk need to update `ArgumentMultimap::containsMultiplePrefix`)

### CODE REVIEW

**Comment by seanflyyy** [comment]
@seanflyyy need to add test cases and wait for @cadencjk to merge his PR for it to work

File: docs/UserGuide.md [hunk 1]
```diff
@@ -494,15 +494,19 @@ Finds all students with a particular tag.
 Formats: `find t/tagName`
 
 - Only one tag can be searched at each time.
+- The tag is case-sensitive.
```
Comment (Echomo-Xinyu) [inline docs/UserGuide.md:497]: Good catch for including this specification for the behaviour and updating the test case accordingly!

**Review by Echomo-Xinyu — APPROVED** [review]
The implementation looks very clean and good job!

But I wonder whether we should be so strict about the case though? for example, I have "java" and "Java" for tags in my local JSON file. Do we really need to be so strict about the casing of words to such an extent that "java" is different from "Java"?

**Comment by seanflyyy** [comment]
> With the freeze of new features in v1.4, I think this fix is allowed as finding for multiple tags is not clearly disallowed but we should not change the other behaviour of case-insensitive. Therefore, I think it would be better to allow the case insensitivity.

The initial implementation was already case-insensitive

**Comment by Echomo-Xinyu** [comment]
> > With the freeze of new features in v1.4, I think this fix is allowed as finding for multiple tags is not clearly disallowed but we should not change the other behaviour of case-insensitive. Therefore, I think it would be better to allow the case insensitivity.
> 
> The initial implementation was already case-insensitive

Ya that's my point, so we shouldn't change it to case-sensitive now.

**Comment by seanflyyy** [comment]
> > > With the freeze of new features in v1.4, I think this fix is allowed as finding for multiple tags is not clearly disallowed but we should not change the other behaviour of case-insensitive. Therefore, I think it would be better to allow the case insensitivity.
> > 
> > 
> > The initial implementation was already case-insensitive
> 
> Ya that's my point, so we shouldn't change it to case-sensitive now.

Right, my apologies. I misread

**Review by Echomo-Xinyu — COMMENTED** [review]
I think the current implementation will result the application to freeze with the empty tag ie when inputting `find t/`. Can you try to fix it in a later commit?

**Comment by seanflyyy** [comment]
close #307 #309

**Comment by seanflyyy** [comment]
close #270

**Pull request merged**